rust/xaynet-mobile/xaynet_ffi.h

```diff
@@ -7,50 +7,15 @@
 #include <stdint.h>
 #include <stdlib.h>
 /**
- * The maximum number of models to be aggregated.
- */
-enum MASKCONFIG_MODELTYPE {
-  MODELTYPE_M3 = 3, /** At most 1_000 models to be aggregated. */
-  MODELTYPE_M6 = 6, /** At most 1_000_000 models to be aggregated. */
-  MODELTYPE_M9 = 9, /** At most 1_000_000_000 models to be aggregated. */
-  MODELTYPE_M12 = 12, /** At most 1_000_000_000_000 models to be aggregated. */
-};
-
-/**
- * The bounds of the numerical values.
- */
-enum MASKCONFIG_BOUNDTYPE{
-  BOUNDTYPE_B0 = 0, /** Numerical values absolutely bounded by 1 */
-  BOUNDTYPE_B2 = 2, /** Numerical values absolutely bounded by 100 */
-  BOUNDTYPE_B4 = 4, /** Numerical values absolutely bounded by 10_000 */
-  BOUNDTYPE_B6 = 6, /** Numerical values absolutely bounded by 1_000_000 */
-  BOUNDTYPE_BMAX = 255, /** Numerical values absolutely bounded by their original primitive data type's maximum absolute */
-};
-
-/**
- * The original primitive data type of the numerical values to be masked.
+ * The original primitive data type of the model weights.
  */
-enum MASKCONFIG_DATATYPE {
+enum SETMODEL_DATATYPE {
```

Comment:
Later we can i think about getting that information from the client and using it directly in the `xaynet_ffi_participant_set_model` method

```rust
pub unsafe extern "C" fn xaynet_ffi_participant_set_model(
    participant: *mut Participant,
    buffer: *const c_void,
    len: c_uint,
) -> c_int {
    let participant = match unsafe { participant.as_mut() } {
        Some(participant) => participant,
        None => return ERR_NULLPTR,
    };

    let data_type = participant.model_settings().data_type;

    let model = match data_type {
    ...
    }
... 
}

```

wdyt?

Reply:
Good point. I'm going to touch to the area of the code anyway, because we want to make the scalar part of the model store.